src/find_boundaries.jl

```diff
@@ -0,0 +1,303 @@
+"""
+    find_boundaries!(img::AbstractArray, [region,]; background = 0)
+
+Finds the boundaries that are just within each object, replacing the original image.
+`background` is the scalar value of the background pixels which will not be marked as boundaries.
+`region` indicates which dimensions to detect boundaries along.
+
+See `find_boundaries` for examples.
+"""
+function find_boundaries!(img::AbstractArray{Bool,N};
+                         background=false,
+                         kwargs...
+                        ) where N
+    # Find regions where there is at least one background pixel
+    # centered on a foreground pixel
+    if(background != 0)
+        # background_img = img
```

Comment:
Note that this is for the `AbstractArray{Bool}` case, so I was initially expecting background to be a `Bool` itself, either `true` or `false`. However, it appears that `background = - 1` (or any other value that does not equal `true` or `false` may be a natural way to emulate `find_boundaries_thick`..

Reply:
Here is my proposed behavior. I am still contemplating the best implementation.

The implementation I am working on now is if the background is neither equal to `true` or `false`, then I convert the element type of the image to the type of the background and then use the more general version of `find_boundaries`.

Perhaps though the faster version would be to always convert to `Int8` rather than the type of the background which is likely to be `Int64`. Without loss of generality, we can then pass any background that is neither `true` or `false`.

Another implicit optimization is that we know if the background is neither `true` or `false` that the entire image is the foreground.

```julia
julia> B
16×16 BitMatrix:
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  0  1  1  1  1  1  1  1  1  1  0  0  0  0
 0  0  0  1  1  1  1  1  1  1  1  1  0  0  0  0
 0  0  0  1  1  1  1  1  1  1  1  1  0  0  0  0
 0  0  0  1  1  1  1  1  1  1  1  1  0  0  0  0
 0  0  0  1  1  1  1  1  1  1  1  1  0  0  0  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  1  1  1  1  0  0  0  0  0  0  1  1  1  0
 0  0  1  1  1  1  0  0  0  0  0  0  1  1  1  0
 0  0  1  1  1  1  0  0  0  0  0  0  1  1  1  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0

julia> find_boundaries(B) # background = false, inner boundaries
16×16 BitMatrix:
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  0  1  1  1  1  1  1  1  1  1  0  0  0  0
 0  0  0  1  0  0  0  0  0  0  0  1  0  0  0  0
 0  0  0  1  0  0  0  0  0  0  0  1  0  0  0  0
 0  0  0  1  0  0  0  0  0  0  0  1  0  0  0  0
 0  0  0  1  1  1  1  1  1  1  1  1  0  0  0  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  1  1  1  1  0  0  0  0  0  0  1  1  1  0
 0  0  1  0  0  1  0  0  0  0  0  0  1  0  1  0
 0  0  1  1  1  1  0  0  0  0  0  0  1  1  1  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0

julia> find_boundaries(B; background = true) # outer boundaries
16×16 BitMatrix:
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  1  1  1  1  1  1  1  1  1  1  1  0  0  0
 0  0  1  0  0  0  0  0  0  0  0  0  1  0  0  0
 0  0  1  0  0  0  0  0  0  0  0  0  1  0  0  0
 0  0  1  0  0  0  0  0  0  0  0  0  1  0  0  0
 0  0  1  0  0  0  0  0  0  0  0  0  1  0  0  0
 0  0  1  0  0  0  0  0  0  0  0  0  1  0  0  0
 0  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1
 0  1  0  0  0  0  1  0  0  0  0  1  0  0  0  1
 0  1  0  0  0  0  1  0  0  0  0  1  0  0  0  1
 0  1  0  0  0  0  1  0  0  0  0  1  0  0  0  1
 0  1  1  1  1  1  1  0  0  0  0  1  1  1  1  1
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0

julia> find_boundaries(B; background = -1) # thick boundaries
16×16 BitMatrix:
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  1  1  1  1  1  1  1  1  1  1  1  0  0  0
 0  0  1  1  1  1  1  1  1  1  1  1  1  0  0  0
 0  0  1  1  0  0  0  0  0  0  0  1  1  0  0  0
 0  0  1  1  0  0  0  0  0  0  0  1  1  0  0  0
 0  0  1  1  0  0  0  0  0  0  0  1  1  0  0  0
 0  0  1  1  1  1  1  1  1  1  1  1  1  0  0  0
 0  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1
 0  1  1  1  1  1  1  0  0  0  0  1  1  1  1  1
 0  1  1  0  0  1  1  0  0  0  0  1  1  0  1  1
 0  1  1  1  1  1  1  0  0  0  0  1  1  1  1  1
 0  1  1  1  1  1  1  0  0  0  0  1  1  1  1  1
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
```